### DISCUSSION TITLE
stenographer "could not open blockfile"

### DISCUSSION THREAD

**Discussion by DebianGuru** [Security-Onion-Solutions/securityonion#6292] [Unsupported Versions] [answered]
I've been watching some of the log files for SO and noticed the stenographer.log keeps showing:

Thread 0 error tracking "1627815949564048": could not open blockfile "/tmp/stenographer011155325/PKT0/1627815949564048": could not open index for "/tmp/stenographer011155325/PKT0/1627815949564048": invalid index file "/tmp/stenographer011155325/IDX0/1627815949564048" missing versions record: leveldb/table: invalid table (bad magic number)

There are no directories called /tmp/stenographer*********

A ps aux |grep stenographer does show a process:
 /usr/bin/stenotype -v --uid=stenographer --gid=stenographer --filter=(....Long steam of digits and letters........ --threads=1 --dir=/tmp/stenographer011155325 --iface=bond0

This leads me to believe that there SHOULD be a /tmp/stenographer011155325 folder, that isn't being created.   There is a file called /nsm/pcaptmp/stenographer011155325/IDX0/1627815949564048

Seems it should be looking into the /nsm/pcaptmp folder instead of /tmp.
 
Is this message a problem or not?

**Comment by dougburks** [comment, marked as answer]
> Seems it should be looking into the /nsm/pcaptmp folder instead of /tmp.

Docker maps /nsm/pcaptmp on the host to /tmp inside the Docker container, so the steno process sees it at /tmp.

**Reply by DebianGuru** [reply]
Thanks for that answer.  However, my /opt/so/log/stenographer/stenographer.log is still full of errors like:
` Thread 0 error tracking "1627815949564048": could not open blockfile "/tmp/stenographer976970900/PKT0/1627815949564048": could not open index for "/tmp/stenographer976970900/PKT0/1627815949564048": invalid index file "/tmp/stenographer976970900/IDX0/1627815949564048" missing versions record: leveldb/table: invalid table (bad magic number)
`
I'm not sure if this is something to be concerned about or not?

**Reply by dougburks** [reply]
This most likely means that `1627815949564048` is an incomplete file due to Stenographer stopping or box restarting. If it's a small number of files, then it's most likely nothing to be concerned about.
